### ISSUE TITLE
while share file system servcie became abnormal, the milvus can not supply query service.

### ISSUE THREAD

**Issue by yu-hailong** [milvus-io/milvus#4492]
**Please state your issue using the following template and, most importantly, in English.**
1rw  node+ 4ro nodes, no mishards. deployed on K8S
and these 5 nodes share data using share file system.

we put db/ , conf/, logs/ on the share file system.
we found while share file system servcie became abnormal, all 4 ro nodes can not supply query service.

we are told maybe writing logs block query,
So we removed logs from share filesystem into local storage, but the test result did not change(share filesystem service down,  milvus service down).

then , how can we build a high available service using milvus, because share file system may be a  single point of failure   
 

**Describe the bug**

**Steps/Code to reproduce behavior**
Follow this [guide](http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

**Expected behavior**

**Method of installation**
- [ X ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)
0.10.3

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by yhmo** [comment]
Use an USB drive can reproduce this:
1. plug-in a usb drive
2. start milvus with normal mode, specify usb folder as db folder in the server_config.yaml
```
   storage:
      path: [usb drive folder]
```
3. create a collection, insert some vectors
4. stop the milvus server
5. start milvus with read-only mode
6. call search(), data load into cache, return result is correct
7. force eject the usb drive
8. call search() again, the log report error "Collection file doesn't exist: xxxxxx", return result is empty

**Comment by yhmo** [comment]
The reason is in the src/db/Utils.cpp, in the GetCollectionFilePath() method, there is a line code to check the existence of data file path:

```
if (boost::filesystem::exists(parent_path)) {
     ......
}
```

**Comment by yu-hailong** [comment]
does it happen in version 11.X?

**Comment by yhmo** [comment]
11.x meta system is different to 0.10.x. In 0.11.x, it also cannot service if the usb drive is ejected, but the error is different.

**Comment by yhmo** [comment]
In 0.10, we can remove the path existence check to fix this problem.

**Comment by yhmo** [comment]
Ok, correct my words, 0.11.x can work if all segments have been merged and load into cache. I just try the steps with 0.11.1, after eject usb drive, if segment merge task is finished, milvus can service well.

**Comment by yu-hailong** [comment]
I think whatever happens, milvus as a basic service, it should be keep running as much as possible.
I wish while you  try to fix the problem ,  high available and avoid “depends on share filesystem” should be considered

**Comment by yu-hailong** [comment]
by the way, this problem will be fixed in 10.x?

**Comment by yhmo** [comment]
pr4506 fix this issue on 0.10.5

**Comment by yu-hailong** [comment]
we found in version 11.x, the problem exist !!

**Comment by yu-hailong** [comment]
> Ok, correct my words, 0.11.x can work if all segments have been merged and load into cache. I just try the steps with 0.11.1, after eject usb drive, if segment merge task is finished, milvus can service.

we found in version 11.0, the problem exist !!

**Comment by yu-hailong** [comment]
@yhmo

 Can you supply a patch  and manual for upgrading milvus?
in a few days, two problem happened because of share file system.
we can not wait uitil version release formally

**Comment by yu-hailong** [comment]
Can you tell us the detail about how  you fix the problem?
and we want fix it in the version 0.10.3 manullly and install our product ENV.

**Comment by yu-hailong** [comment]
Thank you, but i know little about your source.
if we modify code just like the following in version 0.10.3, is it ok?

![image](https://user-images.githubusercontent.com/17443614/103197712-8b045f00-4921-11eb-832f-fc13dbcdcf8c.png)

**Comment by yhmo** [comment]
@yu-hailong 
The code change only remove path-check, you can modify it like me. I think no problem.

**Comment by yu-hailong** [comment]
I got it, thank you very much @yhmo

**Comment by yu-hailong** [comment]
@yhmo 

I tried to merge the modification into 0.10.3 using https://github.com/milvus-io/milvus/blob/master/INSTALL.md#step-2-start-the-docker-container
We found the problem is resovled, but the metrics sent to prometheus become strange: the cluster_label is miss.

![image](https://user-images.githubusercontent.com/17443614/103756528-72472800-504a-11eb-9cbd-983f9680b980.png)

**Comment by yu-hailong** [comment]
by the way, we found the problem exist in 0.11.0
Do you know how we can fix the problem in  0.11.0 because 0.11.1 is not released yet?

**Comment by yhmo** [comment]
@yu-hailong 
You can try remove the code "if (!std::experimental::filesystem::exists(...)" in the /milvus/core/src/segment/SegmentReader.cpp.
These file path check will return error if milvus cannot find the file.

**Comment by yu-hailong** [comment]
> @yu-hailong
> You can try remove the code "if (!std::experimental::filesystem::exists(...)" in the /milvus/core/src/segment/SegmentReader.cpp.These file path check will return error if milvus cannot find the file. I saw there are three places in this file.

I tried as you suggested, But it does not work.

**Comment by yhmo** [comment]
Yesterday I try the steps on 0.11.1, it works fine when use driver is ejected, search is executed in memory, no error report. What error message you encountered?

```
Use an USB drive can reproduce this:
1. insert a usb drive
2. start milvus with normal mode, specify usb folder as db folder in the server_config.yaml
```
   storage:
      path: [usb drive folder]
```
3. create a collection, insert some vectors
4. stop the milvus server
5. start milvus with read-only mode
6. call search(), data load into cache, return result is correct
7. force eject the usb drive
8. call search() successfully, no error report
```

**Comment by yu-hailong** [comment]
> Yesterday I try the steps on 0.11.1, it works fine when usb driver is ejected, search is executed in memory, no error report. What error message you encountered?

We made the share file system down while searching was executing, and seach error happened, but i did not record the error detail.

Then I stopped the search ,and try to reconnected to milvus, we can not connect milvus successfully.



> 
> ```
> 1. insert a usb drive
> 2. start milvus with normal mode, specify usb folder as db folder in the server_config.yaml
I start milvus with 1node (rw) and 1 node(ro). I did it on milvus 11.0, so the config file is milvus.yaml.
And I made db and wal folder in the share file system.

> 
>    storage:
>       path: [usb drive folder]
> 
> 3. create a collection, insert some vectors
> 4. stop the milvus server
> 5. start milvus with read-only mode
> 6. call search(), data load into cache, return result is correct
> 7. force eject the usb drive
> 8. call search() successfully, no error report
> ```

**Comment by yhmo** [comment]
Since pr4506 fix this issue on 0.10.5, and 0.11.0 is no longer maintained, we close this issue.
